test/e2e/networkingnew.go

```diff
@@ -0,0 +1,358 @@
+			}
+		case strings.Contains(string(body), "no endpoints available"):
+			Logf("Attempt: waiting on service/endpoints")
+		default:
+			Logf("Unexpected response:\n%s", body)
+		}
+		return false, nil
+	}))
+
+	if !passed {
+		if body, err := getDetails(); err != nil {
+			Failf("Timed out.  Major error : Couldn't read service details: %v", err)
+		} else {
+			Failf("Timed out. Service details :\n%s", string(body))
+		}
+	}
+	serviceDoneChannel <- svc.Name
+}
+
+//runNetTest Creates a single pod on each host which serves
+//on a unique port in the cluster.  It then binds a service to
+//that port, so that there are "n" nodes to balance traffic to -
+//finally, each node reaches out to ping every other node in
+//the cluster on the given port.
+//The more ports given, the more services will be spun up,
+//i.e. one service per port.
+//To test basic pod networking, send a single port.
+//To soak test the services, we can send a range (i.e. 8000-9000).
+func runNetTest(timeoutSeconds time.Duration, f *Framework, ports []int, nettestVersion string) {
+	defer GinkgoRecover()
+
+	nodes, err := f.Client.Nodes().List(labels.Everything(), fields.Everything())
+	expectNoError(err, "Failed to list nodes")
+
+	// previous tests may have cause failures of some nodes. Let's skip
+	// 'Not Ready' nodes, just in case (there is no need to fail the test).
+	filterNodes(nodes, func(node api.Node) bool {
+		return isNodeReadySetAsExpected(&node, true)
+	})
+
+	if len(nodes.Items) == 0 {
+		Failf("No Ready nodes found.")
+	}
+	if len(nodes.Items) == 1 {
+		// in general, the test requires two nodes. But for local development, often a one node cluster
+		// is created, for simplicity and speed. (see issue #10012). We permit one-node test
+		// only in some cases
+		if !providerIs("local") {
+			Failf("The test requires two Ready nodes on %s, but found just one.", testContext.Provider)
+		}
+		Logf("Only one ready node is detected. The test has limited scope in such setting. " +
+			"Rerun it with at least two nodes to get complete coverage.")
+	}
+	sem := make(chan string, len(ports))
+	for p := range ports {
+
+		go func(nodes *api.NodeList, port int) {
+
+			var svcname = fmt.Sprintf("nettest-%v", port)
+
+			By(fmt.Sprintf("creating a service named %q in namespace %q", svcname, f.Namespace.Name))
+			svc, err := f.Client.Services(f.Namespace.Name).Create(&api.Service{
+				ObjectMeta: api.ObjectMeta{
+					Name: svcname,
+					Labels: map[string]string{
+						"name": svcname,
+					},
+				},
+				Spec: api.ServiceSpec{
+					Ports: []api.ServicePort{{
+						Protocol:   "TCP",
+						Port:       port,
+						TargetPort: util.NewIntOrStringFromInt(port),
+					}},
+					Selector: map[string]string{
+						"name": svcname,
+					},
+				},
+			})
+
+			if err != nil {
+				Failf("unable to create test service named [%s] %v on port %v", svc.Name, err, port)
+			} else {
+				Logf("************************** Created service successfully [%s]", svc.Name)
```

Comment:
This can be done.  I only have them for easy readability of the "important parts" of a large port test, where scanning parallel logs by eye will be necessary/useful for the non-idea (but important) scenario - that the test is failing or slowing down at large scale.